File: Data Structures and Algorithms/167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.py
# Last updated: 6/4/2026, 6:35:56 PM
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l,r = 0, len(numbers) - 1
        while l < r:
            currentSum = numbers[l] + numbers[r]
            if currentSum > target:
                r -= 1
            elif currentSum < target:
                l += 1
            elif currentSum == target:
                return [l+1,r+1]
        return

File: Data Structures and Algorithms/167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.cpp
// Last updated: 6/4/2026, 6:36:05 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;
        int sum = 0;
        while(l<r)
        {
            sum = numbers[l] + numbers[r];
            if (sum > target)
                r -= 1;
            else if(sum < target)
                l += 1;
            else
                return {l + 1, r + 1};
        }
        return {};
    }
};